### ISSUE TITLE
How to check validation loss? 

### ISSUE THREAD

**Issue by Hari7696** [awslabs/gluon-ts#258]
## Description
I am currently using the deepar package. I am able to see the training loss. But, I am not able to see the validation loss. My model is converging on training loss. But, how will I know when to stop it? If I continue the training by monitoring the training loss, it would result in an over fitted model. What do you suggest

## References
- list reference and related literature
- list known implementations

**Comment by Tarun3679** [comment]
I have the same question too. While training we are able to see the average epooch loss, is it related to training loss or validation loss? How do we know when to stop the training?
 I also observed on training a greater number of epooch's, the loss is converging but the predictions are poor which is a clear indication of overfitting. How do we escape from this kind of problem?

**Comment by davidlkl** [comment]
The class Trainer currently does not implement validation check while training. If this is needed urgently, i think you can have a look at trainer/_base.py, method __call__, which is exactly how the model is trained.

I personally edited this to calculate some metrics like sMAPE for every some epoches.

I am also looking forward to the implementation of this feature.

**Comment by jaheba** [comment]
Thanks for raising the issue.

Indeed, validation loss is something that is on our roadmap and we are happy on any help and/or input on this.

**Comment by lotsoffood** [comment]
Related to this question, is there any way to pull training loss and validation loss (once implemented) during training? I utilize Sacred for tracking experiments, and they have a function ex.log_scalar('your_metric', value, step) where step would be the epoch, and value the training or validation loss. Would love to be able to use Sacred/Omniboard to visualize.
